perfdash/github-configs-fetcher.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2015 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"gopkg.in/yaml.v2"
+)
+
+type githubDirContent struct {
+	Name        string `yaml:"name"`
+	Path        string `yaml:"path"`
+	DownloadUrl string `yaml:"download_url"`
+	Type        string `yaml:"type"`
+	Url         string `yaml:"url"`
+}
+
+/*
+Gets config paths from github directory. It uses github API, which is documented
+here: https://developer.github.com/v3/repos/contents/
+
+Example url: https://api.github.com/repos/pierewoj/test-infra/contents/config/jobs/kubernetes/sig-release/release-branch-jobs
+
+Different branch can be specified by appending "?ref=branch-name" parameter at the end
+of the url.
+*/
+func GetConfigsFromGithub(url string) ([]string, error) {
+	var result []string
+	contents, err := getGithubDirContents(url)
+	if err != nil {
+		return nil, err
+	}
+	for _, c := range contents {
+		// Dirs and non-yaml files are ignored; this means that there is no
+		// recursive search, it should be good enough for now.
+		if c.Type == "file" && strings.HasSuffix(c.Name, ".yaml") {
+			result = append(result, c.DownloadUrl)
+		}
+	}
+	return result, nil
+}
+
+func getGithubDirContents(url string) ([]githubDirContent, error) {
+	fmt.Printf("Downloading github spec from %v\n", url)
+	resp, err := http.Get(url)
+	if err != nil {
+		return nil, fmt.Errorf("error calling github API %s: %v", url, err)
```

Comment:
Nit: Consider using github.com/go-errors/errors instead (they have stacktraces!) :)

Reply:
I'll create a separate issue to use it in perfdash, are you fine with this?

Reply:
Sgtm, thanks

Reply:
https://github.com/kubernetes/perf-tests/issues/929

Reply:
https://github.com/kubernetes/perf-tests/issues/929